core/dbt/clients/jinja.py

```diff
@@ -330,38 +337,34 @@ def get_environment(node=None, capture_macros=False):
     return MacroFuzzEnvironment(**args)
 
 
-def parse(string):
+@contextmanager
+def catch_jinja(node=None) -> Iterator[None]:
     try:
-        return get_environment().parse(str(string))
-
-    except (jinja2.exceptions.TemplateSyntaxError,
-            jinja2.exceptions.UndefinedError) as e:
+        yield
+    except jinja2.exceptions.TemplateSyntaxError as e:
         e.translated = False
-        dbt.exceptions.raise_compiler_error(str(e))
+        raise dbt.exceptions.CompilationException(str(e), node) from e
```

Comment:
It looks like we previously set `e.translated = False` when an `UndefinedError` is raised - do we _not_ want to do that here?

Reply:
Correct! `e.translated` only makes sense in the context of `TemplateSyntaxError`s, from my careful reading of the jinja source code. It wasn't a big deal, it was just a useless setattr.